### PR TITLE
[CALCITE-5265] Select operator' parentheses should be same with Union operator

### PR DESCRIPTION
**Pull request by l4wei**
remove necessary parentheses on select clause.

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java
Comment (libenchao) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:6358]: Are there any tests which cover what Julian has mentioned in the JIRA comment?

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java
Comment (libenchao) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:6357]: The comment should also be adapted according to the changes in JIRA title.

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java
Comment (l4wei) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:6358]: There are many unit tests cover "insert into aTable select ? from bTable" in SqlParserTest.java. And RelToSqlConverterTest::testInsertValuesWithDynamicParams will cover "insert into aTable select ? from bTable union all select ? from cTable" mentioned by Julian in JIRA comment.

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java
Comment (l4wei) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: OK, done.

**Review by libenchao — APPROVED** [review]
@l4wei Thanks the updating, the PR LGTM now, merging.

**Comment by libenchao** [comment]
@l4wei I didn't notice Julian has another comment when I commented with "LGTM", could you add another test which addresses Julian's comment?

**Comment by l4wei** [comment]
> 

OK

**Comment by l4wei** [comment]
> 

I have added test that mentioned by Julian' last comment in third commit in this PR. FYI

**Review by libenchao — APPROVED** [review]
LGTM, will merge in 24 hours if there is no more objections appear.

**Pull request closed without merging**
